src/pkg/boltz.ts

```diff
@@ -0,0 +1,480 @@
+  expectedAmount: number;
+  id: string;
+  redeemScript: string;
+}
+
+export interface ReverseSwap {
+  blindingPrivateKey: string;
+  claimPublicKey: string;
+  id: string;
+  invoice: string;
+  lockupAddress: string;
+  preimage: Buffer;
+  redeemScript: string;
+  timeoutBlockHeight: number;
+}
+
+export interface MakeClaimTransactionParams {
+  utxo: Unspent;
+  claimKeyPair: ECPairInterface;
+  preimage: Buffer;
+  redeemScript: Buffer;
+  destinationScript: Buffer;
+  fee: number;
+  blindingPublicKey: Buffer;
+  timeoutBlockHeight: number;
+}
+
+const feeAmount = 500; // fee for regular liquid tx
+const swapFeeAmount = 500; // fee for Boltz
+// lightning swap invoice amount limit (in satoshis)
+export const DEFAULT_LIGHTNING_LIMITS = { maximal: 4294967, minimal: 50000 };
+export const DEPOSIT_LIGHTNING_LIMITS = {
+  maximal: DEFAULT_LIGHTNING_LIMITS.maximal - feeAmount - swapFeeAmount,
+  minimal: DEFAULT_LIGHTNING_LIMITS.minimal - feeAmount - swapFeeAmount,
+};
+
+export const boltzUrl: Record<NetworkString, string> = {
+  regtest: 'http://localhost:9090',
+  testnet: 'https://testnet.boltz.exchange/api',
+  liquid: 'https://api.boltz.exchange',
+};
+
+export class Boltz implements BoltzInterface {
+  private asset: string;
+  private url: string;
+  private zkp: ZKP;
+
+  constructor(url: string, asset: string, zkp: ZKP) {
+    this.asset = asset;
+    this.url = url;
+    this.zkp = zkp;
+  }
+
+  async getBoltzPair(pair: string): Promise<any> {
+    const data = await this.getApi(`${this.url}/getpairs`);
+    if (!data?.pairs?.[pair]) return;
+    return data.pairs[pair];
+  }
+
+  // return invoice expire date
+  getInvoiceExpireDate(invoice: string): number {
+    const toMilliseconds = (num: number) => num * 1000;
+    const { timeExpireDate, timestamp } = bolt11.decode(invoice);
+    if (timeExpireDate) return toMilliseconds(timeExpireDate);
+    const { expire_time } = bolt11.decode(invoice).tagsObject;
+    const expireTime = toMilliseconds(expire_time ?? 3600);
+    const from = timestamp ? toMilliseconds(timestamp) : Date.now();
+    return from + expireTime;
+  }
+
+  // return value in given invoice
+  getInvoiceValue(invoice: string): number {
+    const { satoshis, millisatoshis } = bolt11.decode(invoice);
+    if (satoshis) return fromSatoshi(satoshis, 8);
+    if (millisatoshis) return fromSatoshi(Number(millisatoshis) / 1000, 8);
+    return 0;
+  }
+
+  makeClaimTransaction({
+    utxo,
+    claimKeyPair,
+    preimage,
+    redeemScript,
+    destinationScript,
+    fee,
+    blindingPublicKey,
+    timeoutBlockHeight,
+  }: MakeClaimTransactionParams): Transaction {
+    if (!utxo.blindingData) throw new Error('utxo is not blinded');
+    if (!utxo.witnessUtxo) throw new Error('utxo missing witnessUtxo');
+    const pset = Creator.newPset();
+    const updater = new Updater(pset);
+
+    pset.addInput(
+      new CreatorInput(utxo.txid, utxo.vout, 0xfffffffd, timeoutBlockHeight).toPartialInput()
```

Comment:
The claim transaction does *not* need a timeout block height. It shouldn't have it. Else you can't broadcast until the timeout is reached which is when we refund

The timelock is only needed for refund transactions to make `OP_CHECKLOCKTIMEVERIFY` work

Reply:
Fixed, thanks again for your help.